src/libraries/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.Unix.cs

```diff
@@ -22,7 +22,7 @@ public unsafe abstract partial class FileSystemEnumerator<TResult> : CriticalFin
         private string? _currentPath;
         private IntPtr _directoryHandle;
         private bool _lastEntryFound;
-        private Queue<string>? _pending;
+        private Queue<(string Path, int RemainingDepth)>? _pending;
```

Comment:
```suggestion
        private Queue<(string, int)>? _pending;
```
You don't access these Aliases at all, why do we need them?

Reply:
Only as self code documentation. The code in FileSystemEnumerator.Windows.cs file already follows this pattern.